plumbing/format/packfile/object_pack.go

```diff
@@ -71,11 +77,24 @@ func (o *ObjectToPack) WantWrite() bool {
 	return o.Offset == 1
 }
 
+// SetOriginal sets both Original and saves size, type and hash
+func (o *ObjectToPack) SetOriginal(obj plumbing.EncodedObject) {
+	o.Original = obj
+	o.originalSize = obj.Size()
```

Comment:
check if obj is nil

Reply:
Added checks to SetOriginal and  BackToOriginal. The test now also uses it to set Original to nil.